aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -166,6 +166,15 @@ Tensor empty_like(const Tensor& self, const TensorOptions& options) {
   return at::empty(self.sizes(), options);
 }
 
+Tensor empty_like(const Tensor& self, const TensorOptions& options, MemoryFormat memory_format) {
+  if (memory_format == MemoryFormat::ChannelsLast)
+  {
+    return at::empty_like(self, options).to(MemoryFormat::ChannelsLast);
```

Comment:
It seems to me that `to()` is obligated to do a (wasted) physical transpose here, because `x.to()` doesn't know that the data in `x` isn't important and can't just restride. So this seems like a very inefficient way to implement `empty` with non-standard layout.

Reply:
why isn't the memory format optional?  It seems like something we would want in TensorOptions, but there may be good reasons not to have it in TensorOptions currently.  But TensorOptions are optionals, so it seems like MemoryFormat should be too?